docs/tutorials/configure/pet-photographer.md

```diff
@@ -0,0 +1,984 @@
+              """
+              vision_service attribute
+              is required for a
+              ColorFilterCam component
+              """
+            )
+        return [actual_cam, vision_service]
+
+    def reconfigure(
+        self,
+        config: ComponentConfig,
+        dependencies: Mapping[ResourceName, ResourceBase]
+    ):
+        """Handles attribute reconfiguration"""
+        actual_cam_name = config.attributes.fields[
+          "actual_cam"
+        ].string_value
+        actual_cam = dependencies[
+          Camera.get_resource_name(actual_cam_name)
+        ]
+        self.actual_cam = cast(Camera, actual_cam)
+
+        vision_service_name = config.attributes.fields[
+          "vision_service"
+          ].string_value
+        vision_service = dependencies[
+          Vision.get_resource_name(
+            vision_service_name
+          )
+        ]
+        self.vision_service = cast(
+           Vision,
+           vision_service
+        )
+
+    async def get_properties(
+              self,
+              *,
+              timeout: Optional[float] = None,
+              **kwargs
+    ) -> Camera.Properties:
+        """Returns details about the camera"""
+        return await self.actual_cam.get_properties()
+
+    async def get_image(
+              self,
+              mime_type: str = "",
+              *,
+              extra: Optional[Dict[str, Any]] = None,
+              timeout: Optional[float] = None,
+              **kwargs
+    ) -> Image.Image:
+        """Filters the output of the underlying camera"""
+        img = await self.actual_cam.get_image()
+        if from_dm_from_extra(extra):
+            detections = await self.vision_service.get_detections(img)
+            if len(detections) == 0:
+                raise NoCaptureToStoreError()
+
+        return img
+
+    async def get_images(
+              self,
+              *,
+              timeout: Optional[float] = None,
+              **kwargs
+    ) -> Tuple[
+         List[NamedImage],
+         ResponseMetadata
+         ]:
+        raise NotImplementedError
+
+    async def get_point_cloud(
+              self,
+              *,
+              extra: Optional[Dict[str, Any]] = None,
+              timeout: Optional[float] = None,
+              **kwargs
+    ) -> Tuple[
+         bytes,
+         str
+         ]:
+        raise NotImplementedError
+
+    async def get_geometries(self) -> List[Geometry]:
+        raise NotImplementedError
+```
+
+- The Python SDK simplifies the verification process by exposing the utility function `from_dm_from_extra`, which handles the check for you.
+- For other programming languages, similar utility functions will be exposed to help you check the caller of your filter function.
+  The method for this verification may vary depending on the specific function and programming language.
+  For detailed information, please refer to the SDK documentation relevant to your specific language.
```

Comment:
What do you think about moving this right below the introduction of the utility function and error? (formerly #### Include the filter module requirements) Since this should be the reference for users wanting to make their own filter module, I want it to be clear that they can follow similar logic to create their own... but this might be obvious already, what do you think?